src/heartbeat/heartbeat.c

```diff
@@ -248,36 +267,37 @@ void hb_rx_cb(const uint8_t* data, uint8_t len) {
         if (len != 8) {
             return;
         }
-        if (data[2] != hb_self_id) {
+        if (data[HB_RECEIVER] != hb_self_id) {
             return;
         }
-        
-        // Received ping
-        if (data[0] == 1) {
-            switch (data[1]) {
-                case HB_OBC:
+
+        // Ping Request received
+        if (data[HB_OPCODE] == HB_PING_REQUEST) {
+            switch (data[HB_SENDER]) {
+                case HB_OBC_PLACEHOLDER:
                     hb_send_obc_resp = true;
                     break;
-                case HB_EPS:
-                    hb_send_eps_resp = true;
-                    break;
-                case HB_PAY:
+                case HB_PAY_PLACEHOLDER:
                     hb_send_pay_resp = true;
                     break;
+                case HB_EPS_PLACEHOLDER:
+                    hb_send_eps_resp = true;
+                    break;
                 default:
                     break;
             }
-        } else if (data[0] == 2) {
-            switch (data[1]) {
-                case HB_OBC:
+        // Ping Response received
+        } else if (data[HB_OPCODE] == HB_PING_RESPONSE) {
+            switch (data[HB_SENDER]) {
+                case HB_OBC_PLACEHOLDER:
                     hb_received_obc_resp = true;
                     break;
-                case HB_EPS:
-                    hb_received_eps_resp = true;
-                    break;
-                case HB_PAY:
+                case HB_PAY_PLACEHOLDER:
                     hb_received_pay_resp = true;
                     break;
+                case HB_EPS_PLACEHOLDER:
+                    hb_received_eps_resp = true;
+                    break;
                 default:
                     break;
```

Comment:
You're still missing a way for the device initiating the heartbeat request to save the restart information somewhere. Add global variables to store the most recent received restart reason and restart count, and set those variables here when you receive a response.

Reply:
Should there be unique global variables for each SSM and is there any risk of the data being lost upon reset and then transmitted incorrectly? Or is this already taken care of elsewhere in the software?

Reply:
You don't need unique global variables for each SSM, a common set of variables is fine. Each subsystem is assumed to keep track of its own restart info properly on restart.